src/main/java/uk/gov/hmcts/darts/audio/helper/UnstructuredDataHelper.java

```diff
@@ -0,0 +1,146 @@
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.common.datamanagement.enums.DatastoreContainerType;
+import uk.gov.hmcts.darts.common.entity.AnnotationDocumentEntity;
+import uk.gov.hmcts.darts.common.entity.CaseDocumentEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalLocationTypeEntity;
+import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.MediaEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectRecordStatusEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionDocumentEntity;
+import uk.gov.hmcts.darts.common.enums.SystemUsersEnum;
+import uk.gov.hmcts.darts.common.repository.ExternalLocationTypeRepository;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectRecordStatusRepository;
+import uk.gov.hmcts.darts.common.repository.UserAccountRepository;
+import uk.gov.hmcts.darts.datamanagement.api.DataManagementApi;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static uk.gov.hmcts.darts.common.enums.ExternalLocationTypeEnum.UNSTRUCTURED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+import static uk.gov.hmcts.darts.datamanagement.DataManagementConstants.MetaDataNames.MEDIA_REQUEST_ID;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class UnstructuredDataHelper {
+    private final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    private final DataManagementApi dataManagementApi;
+    private final ObjectRecordStatusRepository objectRecordStatusRepository;
+    private final ExternalLocationTypeRepository externalLocationTypeRepository;
+    private final UserAccountRepository userAccountRepository;
+
+    private static final List<CompletableFuture> jobsList = new ArrayList<>();
+
+    public static List<CompletableFuture> getJobsList() {
+        return jobsList;
+    }
+
+    public boolean createUnstructured(ExternalObjectDirectoryEntity eodEntityToDelete, ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        boolean returnVal = true;
+        UUID uuid = saveToUnstructuredDataStore(eodEntity, binaryData);
+        if (uuid == null) {
+            returnVal = false;
+        } else {
+            saveToDatabase(eodEntity, uuid);
+            removeFromDatabase(eodEntityToDelete);
+        }
+        return returnVal;
+    }
+
+    private UUID saveToUnstructuredDataStore(ExternalObjectDirectoryEntity eodEntity, BinaryData binaryData) {
+        UUID uuid = null;
+        try {
+            Map<String, String> metadata = new HashMap<>();
+            metadata.put(MEDIA_REQUEST_ID, String.valueOf(eodEntity.getMedia().getId()));
```

Comment:
I'd expect the metadata here to be `media_id`?
We really don't want to be confusing it with `media_request_id`...